File: csharp/src/api/Dtos/PatientDto.cs
namespace Example.Api.Dtos;

/// <summary>
/// The data transfer object for a patient.
/// </summary>
/// <value></value>
public record PatientDto
{
    /// <summary>
    /// The unique identifier for the patient.
    /// </summary>
    public long Id { get; init; }

    /// <summary>
    /// Name of the patient.
    /// </summary>
    public string Name { get; init; } = string.Empty;

    /// <summary>
    /// Creation date of the patient record.
    /// </summary>
    public DateTimeOffset CreatedAt { get; init; }

    /// <summary>
    /// The date and time when the patient record was last updated.
    /// </summary>
    /// <value></value>
    public DateTimeOffset? UpdatedAt { get; init; }

    /// <summary>
    /// The patient's orders.
    /// </summary>
    /// <typeparam name="OrderDto"></typeparam>
    /// <returns></returns>
    public ICollection<OrderDto> Orders { get; init; } = new List<OrderDto>();
}

File: src/nodejs/src/dtos/PatientDto.ts
import { OrderDto } from "./OrderDto";

/**
 * Data Transfer Object for patient information.
 */
export class PatientDto {
  /** The unique identifier of the patient. */
  id: number;
  /** The name of the patient. */
  name: string;
  /** The timestamp when the patient was created. */
  createdAt: Date;
  /** The timestamp when the patient was last updated. */
  updatedAt?: Date;
  /** The collection of orders associated with the patient. */
  orders: OrderDto[];

  /**
   * Creates a new patient DTO.
   * @param id - The unique identifier of the patient.
   * @param name - The name of the patient.
   * @param createdAt - The timestamp when the patient was created.
   * @param orders - The collection of orders associated with the patient.
   * @param updatedAt - The timestamp when the patient was last updated.
   */
  constructor(id: number, name: string, createdAt: Date, orders: OrderDto[] = [], updatedAt?: Date) {
    this.id = id;
    this.name = name;
    this.createdAt = createdAt;
    this.orders = orders;
    this.updatedAt = updatedAt;
  }
}
